src/sentry/api/bases/organization_events.py

```diff
@@ -121,21 +110,19 @@ def prev_event_id(self, snuba_args, event):
         Returns the previous event ID if there is a previous event matching the
         conditions provided. Ignores the project_id.
         """
-        conditions = self._apply_start_and_end(snuba_args)
-        prev_event = eventstore.get_prev_event_id(
-            event, conditions=conditions, filter_keys=snuba_args["filter_keys"]
-        )
+        prev_event = eventstore.get_prev_event_id(event, filter=self._get_filter(snuba_args))
 
         if prev_event:
             return prev_event[1]
 
-    def _apply_start_and_end(self, snuba_args):
-        conditions = copy(snuba_args["conditions"])
-        if "start" in snuba_args:
-            conditions.append(["timestamp", ">=", snuba_args["start"]])
-        if "end" in snuba_args:
-            conditions.append(["timestamp", "<=", snuba_args["end"]])
-        return conditions
+    def _get_filter(self, snuba_args):
+        return eventstore.Filter(
+            conditions=snuba_args["conditions"],
+            start=snuba_args.get("start", None),
+            end=snuba_args.get("end", None),
+            project_ids=snuba_args["filter_keys"].get("project_id", None),
+            group_ids=snuba_args["filter_keys"].get("issue", None),
```

Comment:
is snuba_args["filter_keys"].get("project_id", None) going to be a Sequence or an int ?
In other words, are you passing Sequences to the Filter constructor as it expects ?

Reply:
Yes, these filter keys should be sequences already despite the naming convention (https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L701-L707)